### PR TITLE
CI: Acceptance tests at PR v2: enable hw refresh

### PR DESCRIPTION
**Pull request by jordimassaguerpla**
## What does this PR change?

CI: Enable hardware refresh tests

## GUI diff
No difference.
- [X] **DONE**

## Documentation
- No documentation needed
- [X] **DONE**

## Test coverage
- No tests

- [X] **DONE**

## Links

https://github.com/SUSE/spacewalk/issues/20956

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/podman_runner/12_run_salt_sle_minion.sh [hunk 1]
```diff
@@ -2,7 +2,11 @@
...
+echo opensuseminionproductuuid > /tmp/opensuse_product_uuid
+
+podman run --privileged --rm -d --network uyuni-network-1 -v /tmp/opensuse_product_uuid:/sys/class/dmi/id/product_uuid -v /tmp/test-all-in-one:/tmp -v ${src_dir}/testsuite/podman_runner/salt-minion-entry-point.sh:/salt-minion-entry-point.sh --name sle_minion -h sle_minion ghcr.io/$UYUNI_PROJECT/uyuni/ci-test-opensuse-minion:$UYUNI_VERSION bash -c "/salt-minion-entry-point.sh uyuni-server-all-in-one-test 1-SUSE-KEY-x86_64"
 podman exec sle_minion bash -c "ssh-keygen -A && /usr/sbin/sshd -e"
 podman exec sle_minion bash -c "if [ ! -d /root/.ssh ];then mkdir /root/.ssh/;chmod 700 /root/.ssh;fi;cp /tmp/authorized_keys /root/.ssh/"
 
+podman exec sle_minion bash -c "echo DEBUG;cat /sys/class/dmi/id/product_uuid;dmidecode | grep 4c4c4544-0037-3710-8047-c7c04f353632;echo FI_DEBUG"
```
Comment (srbarrios) [inline testsuite/podman_runner/12_run_salt_sle_minion.sh:11]: Don't we need to generate a new one each time? 🤔

**Review by srbarrios — APPROVED** [review]

File: testsuite/Gemfile.lock [hunk 2]
```diff
@@ -0,0 +1,170 @@
+GEM
```
Comment (srbarrios) [inline testsuite/Gemfile.lock:1]: We shouldn't commit this file.

File: testsuite/features/support/constants.rb [hunk 3]
```diff
@@ -129,7 +129,7 @@
 # For containers we do not have SCC, so we set the Fake Base Channel
 # for sle_minion
 sle_base_channel =
-  if ENV['PROVIDER'].include? 'podman'
+  if ENV['PROVIDER'].include?('podman') || ENV['PROVIDER'].include?('docker')
```
Comment (srbarrios) [inline testsuite/features/support/constants.rb:132]: There is a real need in the test framework to distinguish "podman" and "docker"?
Or should we name it "container" as a more generic name?

**Comment by jordimassaguerpla** [comment]
@srbarrios : could you review again?

File: testsuite/Gemfile [hunk 4]
```diff
@@ -29,3 +29,4 @@ gem 'redis', '~> 5.0'
 # or build and install from https://github.com/openSUSE/twopence
 gem 'twopence', "~> 0.4"
 gem 'pg', "~> 1.4"
+gem 'public_suffix', '~> 4.0'
```
Comment (srbarrios) [inline testsuite/Gemfile:32]: Why we need this new gem dependency?

File: testsuite/features/step_definitions/common_steps.rb [hunk 5]
```diff
@@ -103,7 +103,7 @@
     step %(I should see a "#{rounded_uptime_hours} hours ago" text)
   elsif rounded_uptime_minutes >= 45 && rounded_uptime_hours == 1 # shows "an hour ago" from 45 minutes onwards up to 1.5 hours
     step %(I should see a "an hour ago" text)
-  elsif rounded_uptime_minutes > 1 && rounded_uptime_hours < 1
+  elsif rounded_uptime_minutes > 1 && rounded_uptime_hours <= 1
```
Comment (srbarrios) [inline testsuite/features/step_definitions/common_steps.rb:106]: How that is related to enabling Hardware refresh? If that's out of topic, must go in a different PR

File: testsuite/podman_runner/02_setup_network.sh [hunk 6]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 set -ex
-podman network create uyuni-network-1
+sudo -i podman network create --ipv6 --subnet 2001:db8::/64 uyuni-network-1
```
Comment (srbarrios) [inline testsuite/podman_runner/02_setup_network.sh:3]: That might need to be parameterizable maybe.

File: testsuite/podman_runner/03_run_controller.sh [hunk 7]
```diff
@@ -2,6 +2,6 @@
 set -ex
 src_dir=$(cd $(dirname "$0")/../.. && pwd -P)
 
-podman run --rm -d --network uyuni-network-1 -v /tmp/test-all-in-one:/tmp --name controller-test -h controller-test -v ${src_dir}/testsuite:/testsuite ghcr.io/$UYUNI_PROJECT/uyuni/ci-test-controller-dev:$UYUNI_VERSION
+sudo -i podman run --rm -d --network uyuni-network-1 -v /tmp/test-all-in-one:/tmp --name controller-test -h controller-test -v ${src_dir}/testsuite:/testsuite ghcr.io/$UYUNI_PROJECT/uyuni/ci-test-controller-dev:$UYUNI_VERSION
```
Comment (srbarrios) [inline testsuite/podman_runner/03_run_controller.sh:5]: Not sure about it. But maybe is clear no run the scripts (.sh files) using sudo, so you don't need to worry inside the script.

File: testsuite/podman_runner/10_setup_sshd.sh [hunk 8]
```diff
@@ -1,9 +1,9 @@
...
 set -xe
 
-podman exec uyuni-server-all-in-one-test bash -c "ssh-keygen -A && /usr/sbin/sshd -e"
-podman exec uyuni-server-all-in-one-test bash -c "if [ ! -d /root/.ssh ];then mkdir /root/.ssh/;chmod 700 /root/.ssh;fi;cp /tmp/authorized_keys /root/.ssh/"
-podman exec opensusessh bash -c "echo 'root:linux' | chpasswd && echo 123456789 > /etc/machine-id"
-podman exec opensusessh bash -c "if [ ! -d /root/.ssh ];then mkdir /root/.ssh/;chmod 700 /root/.ssh;fi;cp /tmp/authorized_keys /root/.ssh/"
-podman exec controller-test bash -c "if [ ! -d /root/.ssh ];then mkdir /root/.ssh/;chmod 700 /root/.ssh;fi;cp /tmp/authorized_keys /root/.ssh/"
+sudo -i podman exec uyuni-server-all-in-one-test bash -c "ssh-keygen -A && /usr/sbin/sshd -e"
+sudo -i podman exec uyuni-server-all-in-one-test bash -c "if [ ! -d /root/.ssh ];then mkdir /root/.ssh/;chmod 700 /root/.ssh;fi;cp /tmp/authorized_keys /root/.ssh/"
+sudo -i podman exec opensusessh bash -c "echo 'root:linux' | chpasswd && echo 123456789 > /etc/machine-id"
```
Comment (srbarrios) [inline testsuite/podman_runner/10_setup_sshd.sh:6]: Can we have a machine-id with that format? Wouldn't that affect in any way?

File: testsuite/podman_runner/12_run_salt_sle_minion.sh [hunk 9]
```diff
@@ -4,7 +4,10 @@ src_dir=$(cd $(dirname "$0")/../.. && pwd -P)
...
-podman exec sle_minion bash -c "ssh-keygen -A && /usr/sbin/sshd -e"
-podman exec sle_minion bash -c "if [ ! -d /root/.ssh ];then mkdir /root/.ssh/;chmod 700 /root/.ssh;fi;cp /tmp/authorized_keys /root/.ssh/"
+sudo -i podman run --privileged --rm -d --network uyuni-network-1 -v /tmp/opensuse_product_uuid:/sys/class/dmi/id/product_uuid -v /tmp/test-all-in-one:/tmp -v ${src_dir}/testsuite/podman_runner/salt-minion-entry-point.sh:/salt-minion-entry-point.sh --name sle_minion -h sle_minion ghcr.io/$UYUNI_PROJECT/uyuni/ci-test-opensuse-minion:$UYUNI_VERSION bash -c "/salt-minion-entry-point.sh uyuni-server-all-in-one-test 1-SUSE-KEY-x86_64"
+sudo -i podman exec sle_minion bash -c "ssh-keygen -A && /usr/sbin/sshd -e"
+sudo -i podman exec sle_minion bash -c "if [ ! -d /root/.ssh ];then mkdir /root/.ssh/;chmod 700 /root/.ssh;fi;cp /tmp/authorized_keys /root/.ssh/"
+
```
Comment (srbarrios) [inline testsuite/podman_runner/12_run_salt_sle_minion.sh:10]: The extra new lines can be removed.

File: testsuite/podman_runner/14_run_salt_deblike_minion.sh [hunk 10]
```diff
@@ -4,6 +4,7 @@ src_dir=$(cd $(dirname "$0")/../.. && pwd -P)
...
-podman exec deblike_minion bash -c "ssh-keygen -A && /usr/sbin/sshd -e"
-podman exec deblike_minion bash -c "if [ ! -d /root/.ssh ];then mkdir /root/.ssh/;chmod 700 /root/.ssh;fi;cp /tmp/authorized_keys /root/.ssh/"
+sudo -i podman run --privileged --rm -d --network uyuni-network-1 -v /tmp/ubuntu_product_uuid:/sys/class/dmi/id/product_uuid -v /tmp/test-all-in-one:/tmp -v ${src_dir}/testsuite/podman_runner/salt-minion-entry-point.sh:/salt-minion-entry-point.sh --name deblike_minion -h deblike_minion ghcr.io/$UYUNI_PROJECT/uyuni/ci-test-ubuntu-minion:$UYUNI_VERSION bash -c "/salt-minion-entry-point.sh uyuni-server-all-in-one-test 1-DEBLIKE-KEY"
+sudo -i podman exec deblike_minion bash -c "ssh-keygen -A && /usr/sbin/sshd -e"
+sudo -i podman exec deblike_minion bash -c "if [ ! -d /root/.ssh ];then mkdir /root/.ssh/;chmod 700 /root/.ssh;fi;cp /tmp/authorized_keys /root/.ssh/"
+
```
Comment (srbarrios) [inline testsuite/podman_runner/14_run_salt_deblike_minion.sh:10]: extra line

File: testsuite/Gemfile (hunk 4 above)
Comment (jordimassaguerpla) [reply testsuite/Gemfile:32]: It is a dependency already, but we need it to be ~> 4.0 or it pulls a newer version which depends on a newer ruby version which opensuse 15.4 does not have.

File: testsuite/features/step_definitions/common_steps.rb (hunk 5 above)
Comment (jordimassaguerpla) [reply testsuite/features/step_definitions/common_steps.rb:106]: This makes the test succeed :)  . If you have an uptime of 35 minutes, when you rounded is 1h. Without this fix, the test was looking for 0 days instead.

File: testsuite/podman_runner/03_run_controller.sh (hunk 7 above)
Comment (jordimassaguerpla) [reply testsuite/podman_runner/03_run_controller.sh:5]: Then I would need to put the sudo in each step of the github action..I thought about it , too.

File: testsuite/podman_runner/02_setup_network.sh (hunk 6 above)
Comment (jordimassaguerpla) [reply testsuite/podman_runner/02_setup_network.sh:3]: Maybe, the thing is that I do not understand very much which subnets can you use...

File: testsuite/podman_runner/10_setup_sshd.sh (hunk 8 above)
Comment (jordimassaguerpla) [reply testsuite/podman_runner/10_setup_sshd.sh:6]: Tests pass.... I guess if tests failed we could think about it.

File: testsuite/podman_runner/12_run_salt_sle_minion.sh (hunk 1 above)
Comment (jordimassaguerpla) [reply testsuite/podman_runner/12_run_salt_sle_minion.sh]: Yep. I have removed this line from the code.

File: testsuite/Gemfile.lock (hunk 2 above)
Comment (jordimassaguerpla) [reply testsuite/Gemfile.lock]: Agreed. I have removed it and instead fix the dependencies in the Gemfile.

File: testsuite/features/support/constants.rb (hunk 3 above)
Comment (jordimassaguerpla) [reply testsuite/features/support/constants.rb]: Not anymore. We only support podman, so I have removed the "include docker" thing.

File: testsuite/podman_runner/12_run_salt_sle_minion.sh (hunk 9 above)
Comment (jordimassaguerpla) [reply testsuite/podman_runner/12_run_salt_sle_minion.sh:10]: done

File: testsuite/podman_runner/14_run_salt_deblike_minion.sh (hunk 10 above)
Comment (jordimassaguerpla) [reply testsuite/podman_runner/14_run_salt_deblike_minion.sh:10]: done

File: testsuite/Gemfile (hunk 4 above)
Comment (nodeg) [reply testsuite/Gemfile:32]: See https://github.com/weppos/publicsuffix-ruby/blob/main/CHANGELOG.md#500

File: testsuite/podman_runner/02_setup_network.sh (hunk 6 above)
Comment (jordimassaguerpla) [reply testsuite/podman_runner/02_setup_network.sh]: I just added the optional parameter "s" for setting a different subnet. If the parameter is not set, it uses the default 2001:db8::/64

File: testsuite/features/step_definitions/common_steps.rb (hunk 5 above)
Comment (nodeg) [reply testsuite/features/step_definitions/common_steps.rb:106]: Please remove this change then here once the other PR is merged.

**Review by nodeg — APPROVED** [review]
